File: 2015-01/python/part2.py
def run(file):
    line = file.readline()
    floor = 0
    for pos, char in enumerate(line, 1):
        floor += 1 if char == "(" else -1
        if floor == -1:
            return pos


if __name__ == "__main__":
    with open("../input.txt", 'r') as f:
        print(run(f))

File: 2015-01/rust/part2.rs
use std::fs;

fn main() {
    let filename = "../input.txt";
    let contents = fs::read_to_string(filename).expect("Should have been able to read the file");

    let mut floor = 0;
    for (i, c) in contents.chars().enumerate() {
        floor += if c == '(' { 1 } else { -1 };
        if floor == -1 {
            println!("{}", i + 1);
            break;
        }
    }
}
